### PR TITLE
Add collection integration tests

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
This doesn't fully work (even running locally, you can forget about CI). But the reasons it doesn't work are kind of the point.

Firstly, the `tower_job_template` module fails because https://github.com/ansible/awx/pull/5173 needs to be merged first.

Next, the `tower_project` doesn't work on the first run. That needs https://github.com/ansible/awx/issues/5166. This issue gets at the heart of why I'm working on this. There is no way to develop a fix for that with the unit tests. That's expected, some problems only emerge with all the parts are brought together, and unit tests just won't cut it.

Let me clarify that I do not want an expansive set of integration tests for the collection. I would prefer this to be the absolute minimum needed for clear communication and ability to address the problems that primary development of the collection faces.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - AWX collection

##### AWX VERSION
```
9.0.0
```

##### ADDITIONAL INFORMATION
Current status:

```
$ make test_collection_integration
rm -rf sanity
mkdir -p sanity/ansible_collections/awx
cp -Ra awx_collection sanity/ansible_collections/awx/awx
cd sanity/ansible_collections/awx/awx && git init && git add .
Initialized empty Git repository in /Users/alancoding/Documents/tower/sanity/ansible_collections/awx/awx/.git/
cd sanity/ansible_collections/awx/awx && ansible-test integration demo
WARNING: Ignoring tests in "test/" in favor of "tests/".
WARNING: A Python 2 pip was found at "/usr/local/bin/pip2", but it uses interpreter "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python" instead of "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/bin/python2.7".
WARNING: A Python 2.7 pip was found at "/usr/local/bin/pip2.7", but it uses interpreter "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python" instead of "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/bin/python2.7".
Running demo integration test script
[WARNING]: running playbook inside collection awx.awx

[WARNING]: The `junit_xml` python module is not installed. Disabling the `junit` callback plugin.


PLAY [localhost] ********************************************************************************************************************************************************************

TASK [awx.awx.tower_organization] ***************************************************************************************************************************************************
ok: [localhost]

TASK [awx.awx.tower_inventory] ******************************************************************************************************************************************************
ok: [localhost]

TASK [awx.awx.tower_host] ***********************************************************************************************************************************************************
ok: [localhost]

TASK [awx.awx.tower_project] ********************************************************************************************************************************************************
ok: [localhost]

TASK [awx.awx.tower_credential_type] ************************************************************************************************************************************************
changed: [localhost]

TASK [awx.awx.tower_credential] *****************************************************************************************************************************************************
changed: [localhost]

TASK [awx.awx.tower_credential] *****************************************************************************************************************************************************
changed: [localhost]

TASK [awx.awx.tower_job_template] ***************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Unsupported parameters for (awx.awx.tower_job_template) module: credentials Supported parameters include: ask_credential, ask_diff_mode, ask_extra_vars, ask_inventory, ask_job_type, ask_limit, ask_skip_tags, ask_tags, ask_verbosity, become_enabled, concurrent_jobs_enabled, credential, description, diff_mode_enabled, extra_vars_path, fact_caching_enabled, force_handlers_enabled, forks, host_config_key, inventory, job_tags, job_type, limit, name, playbook, project, skip_tags, start_at_task, state, survey_enabled, survey_spec, timeout, tower_config_file, tower_host, tower_password, tower_username, validate_certs, vault_credential, verbosity"}

PLAY RECAP **************************************************************************************************************************************************************************
localhost                  : ok=7    changed=3    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

NOTICE: To resume at this test target, use the option: --start-at demo
ERROR: Command "./runme.sh" returned exit status 2.
make: *** [test_collection_integration] Error 1
```

(it will fail on `tower_project`, but this is the 2nd running, so project already exists)

### CODE REVIEW

File: awx_collection/tests/integration/targets/demo/test_demo.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        env_value: "use of custom input is confirmed"
+
+  - awx.awx.tower_job_template:
+      name: demo
+      project: demo
+      inventory: demo
+      playbook: awx_collection/tests/integration/targets/zen/test_job_run.yml
+      credentials:
+        - demo_vault
+        - demo_custom
```
Comment (AlanCoding) [inline awx_collection/tests/integration/targets/demo/test_demo.yml:56]: @ryanpetrello thought you might like this syntax

File: awx_collection/tests/integration/targets/demo/test_demo.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      name: demo
+      variables: "ansible_connection: local"
+      inventory: demo
+
+  - awx.awx.tower_project:
+      name: demo
+      organization: demo
+      scm_type: git
+      scm_url: https://github.com/AlanCoding/awx.git
+      scm_branch: collection_integration
```
Comment (AlanCoding) [inline awx_collection/tests/integration/targets/demo/test_demo.yml]: remove and point to ansible/awx if merge is about to happen

**Comment by AlanCoding** [comment]
Some additional content to add in at the end of the playbook:

```yaml
  - awx.awx.tower_job_launch:
      job_template: demo
    register: job

  - awx.awx.tower_job_wait:
      job_id: "{{ job.id }}"
    register: job_output

  - assert:
      that: "job_output.status == 'successful'"
```

also, according to the linked issue, we could consider adding unicode to `awx_collection/tests/integration/targets/zen/test_job_run.yml`. But I'm still split on that particular issue. Maybe we should, maybe we shouldn't.

**Review by chrismeyersfsu — APPROVED** [review]

**Review by beeankha — APPROVED** [review]
🚀

File: awx_collection/README.md [hunk 3]
```diff
@@ -71,6 +71,19 @@ pip install -e .
...
+The old integration tests are kept passing, but not being expanded on.
+A master copy is hosted at: https://github.com/ansible/test-playbooks/tree/master/tower_modules
+
+A new set of integration tests as a monolithic demo is available as
+"demo" and "remove" in the `tests/integration/targets` folder. This is intended to
+server a dual-purpose as a tl;dr summary of how to use the modules.
+
+Inside the development container `dnf install ansible-test` is needed.
+Authentication setup is needed, `tower-cli login <username>`.
+Then run `make test_collection_integration`.
```
Comment (AlanCoding) [inline awx_collection/README.md:89]: Also need to call out `pip install junit-xml`

Comment (AlanCoding) [reply awx_collection/README.md:89]: @squidboylan right now this is more of a manual recipe, but we can get something like this into automation. This PR has usable stuff IMO, but I don't want to merge it without a more clear idea about where stuff will live. We have several workable options, I just want to have it all consolidated and clear.

File: awx_collection/README.md [hunk 4]
```diff
@@ -71,6 +71,23 @@ pip install -e .
 PYTHONPATH=awx_collection:$PYTHONPATH py.test awx_collection/test/awx/
 ```
 
+### Integration
+
+The old integration tests are kept passing, but not being expanded on.
```
Comment (beeankha) [inline awx_collection/README.md]: Nitpick: That sentence should read "The old integration tests are passing, ..." (remove "kept")

**Comment by AlanCoding** [comment]
there's more stuff we need to do before we can actually connect the wires here.

**Pull request closed without merging**